lib/datadog/core/telemetry/client.rb

```diff
@@ -41,7 +41,7 @@ def disable!
         end
 
         def started!
-          return if !@enabled || self.class.started
+          return if !@enabled || self.class.started || forked?
```

Comment:
@jennchenn, could you add tests to cover this commit? It seems like no tests broke with this change, and it's an important change to be covered by tests.